examples/loki/nurf_data.py

```diff
@@ -102,120 +102,173 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
-
         # UV subgroup
         grp_uv = hf.create_group("/entry/instrument/uv")
-        grp_uv.attrs["NX_class"] = 'NXdetector'
-        grp_uv.create_dataset('description', data='UV')
-        grp_uv.create_dataset('type',
-                              data='HR4PRO-UV-VIS-ES, DH-2000-FHS-DUV-TTL, QP600-2-SR')
-    
-        # create the various datasets for UV
-        uv_inttime = grp_uv.create_dataset('uv_integration_time',
-                                           data=data['UV_IntegrationTime'],
-                                           shape=data[
-                                               'UV_IntegrationTime'].shape,
-                                           dtype=np.int32)
-        uv_inttime.attrs['long_name'] = 'uv_integration_time'
-        uv_inttime.attrs['units'] = 's'  # TODO: unit to be verified
-
-        uv_bkg = grp_uv.create_dataset('uv_background',
-                                       data=data['UV_background'],
-                                       shape=data['UV_background'].shape,
-                                       dtype=np.float32)
-        uv_bkg.attrs['long name'] = 'uv_background'
-
-        uv_int0 = grp_uv.create_dataset('uv_intensity0',
-                                        data=data['UV_intensity0'],
-                                        shape=data['UV_intensity0'].shape,
-                                        dtype=np.float32)
-        uv_int0.attrs['long name'] = 'uv_intensity0'
-
-        uv_spectra = grp_uv.create_dataset('uv_spectra',
-                                           data=data['UV_spectra'],
-                                           shape=data['UV_spectra'].shape,
-                                           dtype=np.float32)
-        uv_spectra.attrs['long name'] = 'uv_spectra'
-
-        uv_wavelength = grp_uv.create_dataset('uv_wavelength',
-                                              data=data['UV_wavelength'],
+        grp_uv.attrs["NX_class"] = 'NXdata'
+        
+        # subgroup for uv all data (sample, dark, reference)
+        uv_signal=grp_uv.create_group("uv_all_data")
+        uv_signal_data=uv_signal.create_dataset('data', data=uv_all_data,
+                                           shape=uv_all_data.shape)
+        uv_signal_data.attrs['long_name']= 'uv_all_data'
+        uv_signal_data.attrs['units']= ''
+        uv_signal_data.attrs['signal']= 'data'  #indicate that the main signal is data 
+        uv_signal_data.attrs['axes']= [ "wavelength", "." ]   #see example in NXdata, time as x-axis is first entry
+        uv_signal_image_key=uv_signal.create_dataset('image_key',data=uv_image_key,shape=uv_image_key.shape, dtype=np.int32)
+       
+     
+        # uv_wavelength
+        uv_wavelength_data=grp_uv.create_dataset('uv_wavelength', data=data['UV_wavelength'],
                                               shape=data['UV_wavelength'].shape,
                                               dtype=np.float32)
```

Comment:
This is references by the `axes` attribute of the signal, so it should be in the same group? Why is the signal in a subgroup of NXdata?

Reply:
I removed the subgroup for the data spectrum. `data`, `image_key ` are out of the subgroup. `uv_wavelength` is still a dataset.